Shared/ImageLoadingHandler.swift

```diff
@@ -0,0 +1,186 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0
+
+import Foundation
+import Kingfisher
+import UIKit
+
+public enum ImageLoadingError: Error, CustomStringConvertible {
+    case unableToFetchImage
+    case iconUrlNotFound
+    case sizeTooLarge
+
+    public var description: String {
+        switch self {
+        case .unableToFetchImage: return "Favicon Image Error: Unable to fetch image"
+        case .iconUrlNotFound: return "Facicon url not found"
+        case .sizeTooLarge: return "Image download size too large"
+        }
+    }
+}
+
+public struct ImageLoadingConstants {
+    public static let MaximumFaviconSize = 1 * 1024 * 1024 // 1 MiB file size limit
+    public static let ExpirationTime = TimeInterval(60*60*24*7) // Only check for icons once a week
+    public static let NoLimitImageSize = 0 // 0 Means there is no limit for image size
+}
+
+protocol ImageFetcher {
```

Comment:
@electricRGB may I ask why not? I'd be happy to modify but would like to understand the reasoning behind it :)

Reply:
I'm just adding my answer here for completion's sake - as this is two days later, lol. 

Essentially, because static methods belong to the class, rather than an instance, testing becomes difficult because mocking them is difficult. I'm not an expert at this, and only remember what @OrlaM was talking about in her presentation but, from what I understood (and given my understanding of coding), we can categorize methods into two types of functions when it comes to (all functions really) static functions: functional functions and unfunctional functions.

Functional functions are fine to be `static` because, given an input, they will always give you the exact same output - testing them is easy. Whereas, unfuctional functions mutate global state, or proxies to a global object, or have some other behaviour that makes them difficult to test.

Reply:
Roux is spot on. If we want to write tests for the object that is using this functionality then we would have no way to mock away this logic if it was static.